worlds/hk/__init__.py

```diff
@@ -165,75 +167,161 @@ def white_palace_exclusions(self):
+                    item2.classification = ItemClassification.progression
+                    if randomized:
+                        pool.append(item1)
+                        pool.append(item2)
                     else:
-                        self.create_location(location_name)
+                        loc1.place_locked_item(item1)
+                        loc2.place_locked_item(item2)
+                elif item_name == "Shade_Cloak" and self.world.SplitMothwingCloak[self.player]:
+                    loc = self.create_location(location_name)
+                    if self.world.random.randint(0, 1):
+                        item = self.create_item("Left_Mothwing_Cloak")
+                    else:
+                        item = self.create_item("Right_Mothwing_Cloak")
+                    item.classification = ItemClassification.progression
+                    if randomized:
                         pool.append(item)
-            # elif option_key not in logicless_options:
-            else:
-                for item_name, location_name in zip(option.items, option.locations):
-                    if location_name in wp_exclusions and location_name != 'King_Fragment':
-                        continue
+                    else:
+                        loc.place_locked_item(item)
+                else:
                     item = self.create_item(item_name)
+                # self.create_location(location_name).place_locked_item(item)
                     if location_name == "Start":
+                        if (item_name == "Focus" and self.world.RandomizeFocus[self.player]) or (item_name == "Swim"
+                                and self.world.RandomizeSwim[self.player]) or (item_name in ["Upslash","Leftslash",
+                                "Rightslash"] and self.world.RandomizeNail[self.player]):
+                            continue
                         self.world.push_precollected(item)
                     else:
-                        self.create_location(location_name).place_locked_item(item)
+
+                        location = self.create_location(location_name)
+                        if not vanilla and location_name in wp_exclusions:
+                            if location_name == 'King_Fragment':
+                                excluded = True
+                            else:
+                                vanilla = True
+                        if excluded:
+                            location.progress_type = LocationProgressType.EXCLUDED
+                        if vanilla:
+                            location.place_locked_item(item)
+                        else:
+                            pool.append(item)
+
+                if location_name in wp_exclusions:
+                    print(f"World {self.world.player_name[self.player]} - Item {item_name} - Location {location_name} - vanilla={vanilla!r}, excluded={excluded!r}")
+
+        for i in range(8, self.world.SlyShopSlots[self.player].value):
```

Comment:
I feel like this could be done in a nested loop rather than multiple copy-pasted for loops.  Also, `self.created_multi_locations["Sly"]` will have the number of pre-existing Sly locations (which should be 6 by now), so that simplifies things.  See also my comments on `create_shop_item`

Something like this may work:

```python
shop_slot_options = { "Sly": "SlyShopSlots" }  # etc.  Should maybe be in Options.py?
   for location, option in shop_slot_options.items():
       for _ in range(self.created_multi_locations[location], getattr(self.world, option).value):
           self.create_location(location)
```

(Another option would be to have the associated shop location name be an attribute on the associated `Option` subclasses)

Relying on self.created_multi_locations will only work if this is done *after* the shops populate just below where this code block currently is.